pylabrobot/resources/carrier.py

```diff
@@ -240,8 +238,10 @@ def get_plate_sinking_depth(plate: Plate):
         z_sinking_depth = get_plate_sinking_depth(first_child)
       resource.register_did_assign_resource_callback(self._update_resource_stack_location)
       self.register_did_unassign_resource_callback(self._deregister_resource_stack_callback)
+    return -Coordinate(z=z_sinking_depth)
```

Comment:
_get_sinking_depth should not handle callbacks

Reply:
made an issue for this: https://github.com/PyLabRobot/pylabrobot/issues/246

Reply:
i'll add the issue to the code in a comment as well